### PR TITLE
423: Block delete of vertex in 4-vertex polygon

### LINKED ISSUE #423: [Draw] Deleting a polygon vertex when there are only 3 causes error, breaks app
To replicate: 
- Draw a square with 4 points 
- Delete one of the points to make a triangle (delete works fine)
- Delete another of the points 
- The app goes into an unusable state with error:  `Uncaught Error: coordinates must only contain numbers`

### PR DESCRIPTION
**Pull request by julialucyhogan**
Also enables delete of multiple points, which are previously multiply
selectable, but only singularly delete-able.

need help figuring out how to access polygon annotation vertex IDs instead of just the whole geometry ID in order to test. 

closes #423

### CODE REVIEW

**Comment by julialucyhogan** [comment]
@hannahkates thank you for testing! I completely forgot to test this in manual draw mode for dev site. Checking now!

**Comment by allthesignals** [comment]
@julialucyhogan I can't reproduce - I tried to do this in one of the rezoning types, didn't see anything specific to vertex deletion.

**Comment by allthesignals** [comment]
Let's get this merged

File: app/components/project-geometries/modes/draw.js [hunk 1]
```diff
@@ -115,18 +118,40 @@ export default class DrawComponent extends Component {
...
+      draw.changeMode('simple_select_delete', { featureIds: [selectedFeature.id] });
+    }
 
-    if (feature) {
-      draw.trash();
-    } else {
-      draw.delete(selectedFeature);
+    // if user selects entire Feature(s) (polygon, line, or point(s)) for deletion
+    // then switch to simple_select_delete mode (cannot delete entire Feature in direct_select mode)
+    if (!selectedFeaturePoint && !['simple_select', 'simple_select_delete'].includes(draw.getMode())) {
```
Comment (allthesignals) [inline app/components/project-geometries/modes/draw.js]: Nice, yes, good call with array literal. It reads nicely in this case.

File: app/components/mapbox-gl-draw.js [hunk 2]
```diff
@@ -32,8 +32,10 @@ export const DefaultDraw = MapboxDraw.bind(null, {
     'draw_annotations:linear': AnnotationsMode, // These are identical because they function the same
     'draw_annotations:curved': AnnotationsMode, // but only really need to be named differently
     'draw_annotations:square': AnnotationsMode,
-    'draw_annotations:label': AnnotationsDrawPoint,
-    'draw_annotations:centerline': AnnotationsDrawPoint,
+    'draw_annotations:label': AnnotationsDrawPointMode,
+    'draw_annotations:centerline': AnnotationsDrawPointMode,
+    // duplicate mode with distinct name  to avoid `skipToDirectSelect` trigger when we switch to simple_select for delete
+    simple_select_delete: MapboxDraw.modes.simple_select,
```
Comment (allthesignals) [inline app/components/mapbox-gl-draw.js]: Just a general thought: are we abusing the custom modes feature here? It's really used as a way to determine _how_ something was produced without the mode itself having anything to do with the output it was given (for example, AnnotationsMode is the same for linear, curved, and square, which are all rendered later based on the "imprinted" mode you see in the hash).

**Pull request merged**